libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -595,23 +595,27 @@ size_t IotNetworkAfr_ReceiveUpto( void * pConnection,
         pNetworkConnection->bufferedByteValid = false;
     }
 
-    /* Block and wait for incoming data. */
-    socketStatus = SOCKETS_Recv( pNetworkConnection->socket,
-                                 pBuffer + bytesReceived,
-                                 bufferSize - bytesReceived,
-                                 0 );
-
-    if( socketStatus <= 0 )
-    {
-        IotLogError( "Error %ld while receiving data.", ( long int ) socketStatus );
-    }
-    else
+    if( bufferSize - bytesReceived > 0 )
```

Comment:
Will something bad happen with the past logic of asking for zero bytes from SOCKETS_Recv?

Reply:
It worked in all past cases.  Though it may have depended on the network stack.